frontend/webadmin/modules/uicommonweb/src/main/java/org/ovirt/engine/ui/uicommonweb/models/storage/UploadImageModel.java

```diff
@@ -169,11 +174,26 @@ public void initialize() {
 
                     getDataCenter().setIsChangeable(isChangeable);
                     if (!isChangeable) {
-                        // Set the selected item on the storage domains list.
-                        AsyncDataProvider.getInstance().getStorageDomainById(
-                                new AsyncQuery<>(storageDomain -> {
-                                    getStorageDomain().setItems(Collections.singletonList(storageDomain));
-                                }), limitToStorageDomainId);
+                        // Set the selected item on the storage domains and data centers lists.
+                        Frontend.getInstance().runQuery(
+                                QueryType.GetStorageDomainListById,
+                                new IdQueryParameters(limitToStorageDomainId),
+                                new AsyncQuery<QueryReturnValue>(returnValue -> {
+                            ArrayList<StorageDomain> domains = returnValue.getReturnValue();
+                            if (!domains.isEmpty()) {
+                                StorageDomain storageDomain = domains.get(0);
```

Comment:
which in general may return a list of Storage Domains - it looks that this is a pretty old behavior that is currently changed, so we can consider only one Storage Domain in the result.